rspace/src/test/scala/coop/rchain/rspace/history/HistorySpec.scala

```diff
@@ -100,22 +236,43 @@ class HistorySpec extends FlatSpec with Matchers with OptionValues with InMemory
       case Skip(affix, _) => affix.toSeq.toList shouldBe bytes
       case p              => fail("unknown trie prefix" + p)
     }
+
+  def randomKey(size: Int) = List.fill(size)((Random.nextInt(256) - 128).toByte)
+
+  def generateRandomInsert(size: Int) = List.fill(size)(insert(randomKey(32)))
+  def generateRandomDelete(size: Int) = List.fill(size)(delete(randomKey(32)))
+  def generateRandomDeleteFromInsert(size: Int, inserts: List[InsertAction]) =
+    Random.shuffle(inserts).take(size).map(i => delete(i.key))
+
+  def updateState(state: TrieMap[KeyPath, Blake2b256Hash], actions: List[HistoryAction]) = {
+    actions.map(
+      a =>
+        a match {
+          case InsertAction(key, value) => state.put(key, value)
+          case DeleteAction(key)        => state.remove(key)
+        }
+    )
+    state
+  }
+
 }
 
 object TestData {
 
   implicit def toByteVector(bytes: KeyPath): ByteVector = ByteVector(bytes)
 
-  val _zeros: KeyPath           = List.fill(32)(0).map(_.toByte)
-  val _zerosOnes: KeyPath       = (List.fill(16)(0) ++ List.fill(16)(1)).map(_.toByte)
-  val _31zeros: KeyPath         = List.fill(31)(0).map(_.toByte)
+  val _zeros: KeyPath           = hexKey("00000000000000000000000000000000")
+  val _zerosOnes: KeyPath       = hexKey("00000000000000001111111111111111")
+  val _31zeros: KeyPath         = hexKey("0000000000000000000000000000000")
   def zerosAnd(i: Int): KeyPath = _31zeros :+ i.toByte
   def prefixWithZeros(s: String): KeyPath = {
     val a = List.fill(32 - (s.length))(0).map(_.toByte)
     val b = s.toCharArray.toList.map(_.asDigit).map(_.toByte)
     a ++ b
   }
 
+  def hexKey(s: String) = s.toList.map(k => Integer.parseInt(k.toString, 16).toByte)
```

Comment:
Why not use existing base16 decode function?
```scala
def hexKey(s: String) = Base16.unsafeDecode(s).toList
```

Reply:
https://github.com/rchain/rchain/blob/dev/crypto/src/main/scala/coop/rchain/crypto/codec/Base16.scala#L28
It checked length.

Reply:
This is OK because in hex format byte is represented with two digits. Byte `0` is in hex `00` so to remove one byte two digits must be removed in hex format.

Reply:
They are different actually.
``` 
Base16.unsafeDecode("000100").toList ==List(0,1,0)
s.toList.map(k => Integer.parseInt("000100", 16).toByte) == List(0,0,0,1,0,0)
```
Do you still want it?

Reply:
nvm